cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -98,31 +107,20 @@ func indexNodeByProviderID(obj interface{}) ([]string, error) {
 	return []string{}, nil
 }
 
-func (c *machineController) findMachine(id string) (*Machine, error) {
-	item, exists, err := c.machineInformer.Informer().GetStore().GetByKey(id)
-	if err != nil {
-		return nil, err
-	}
-
-	if !exists {
-		return nil, nil
-	}
-
-	u, ok := item.(*unstructured.Unstructured)
-	if !ok {
-		return nil, fmt.Errorf("internal error; unexpected type: %T", item)
-	}
+func (c *machineController) findMachine(id string) (*unstructured.Unstructured, error) {
+	return c.findResourceByKey(c.machineInformer.Informer().GetStore(), id)
+}
 
-	machine := newMachineFromUnstructured(u.DeepCopy())
-	if machine == nil {
-		return nil, nil
-	}
+func (c *machineController) findMachineSet(id string) (*unstructured.Unstructured, error) {
+	return c.findResourceByKey(c.machineSetInformer.Informer().GetStore(), id)
+}
 
-	return machine, nil
+func (c *machineController) findMachineDeployment(id string) (*unstructured.Unstructured, error) {
+	return c.findResourceByKey(c.machineDeploymentInformer.Informer().GetStore(), id)
 }
 
-func (c *machineController) findMachineDeployment(id string) (*MachineDeployment, error) {
-	item, exists, err := c.machineDeploymentInformer.Informer().GetStore().GetByKey(id)
+func (c *machineController) findResourceByKey(store cache.Store, key string) (*unstructured.Unstructured, error) {
```

Comment:
Need to add test cases for this method to verify that autodiscovery config is doing the right thing.